tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py

```diff
@@ -0,0 +1,207 @@
+
+from typing import Optional
+import logging
+import os
+
+import pandas as pd
+
+from autogluon.core.constants import BINARY, MULTICLASS, REGRESSION
+from autogluon.core.features.types import R_OBJECT, S_IMAGE_URL
+from autogluon.core.models import AbstractModel
+from autogluon.core.utils import get_cpu_count, get_gpu_count, try_import_mxnet, try_import_autogluon_vision
+
+logger = logging.getLogger(__name__)
+
+
+# FIXME: Avoid hard-coding 'image' column name
+# TODO: Handle multiple image columns?
+# TODO: Handle multiple images in a single image column?
+# TODO: Add regression support
+# TODO: refit_full does not work as expected: It won't use all data, will just split train data internally.
+class ImagePredictorModel(AbstractModel):
+    """Wrapper of autogluon.vision.ImagePredictor"""
+    nn_model_name = 'image_nn'
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._label_column_name = 'label'
+        self._load_model = None  # Whether to load inner model when loading.
+        self._classes = None
+        self._internal_feature_map = None
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            get_features_kwargs=dict(
+                valid_raw_types=[R_OBJECT],
+                required_special_types=[S_IMAGE_URL],
+            ),
+        )
+        default_auxiliary_params.update(extra_auxiliary_params)
+        return default_auxiliary_params
+
+    @classmethod
+    def _get_default_ag_args(cls) -> dict:
+        default_ag_args = super()._get_default_ag_args()
+        extra_ag_args = {
+            'valid_stacker': False,
+            'problem_types': [BINARY, MULTICLASS],  # Does not support regression
+        }
+        default_ag_args.update(extra_ag_args)
+        return default_ag_args
+
+    def _preprocess(self, X, fit=False, **kwargs):
+        X = super()._preprocess(X, **kwargs)
+        if fit:
+            X_features = list(X.columns)
+            if len(X_features) != 1:
+                raise AssertionError(f'ImagePredictorModel only supports one image feature, but {len(X_features)} were given: {X_features}')
+            if X_features[0] != 'image':
+                self._internal_feature_map = {X_features[0]: 'image'}
+        if self._internal_feature_map:
+            X = X.rename(columns=self._internal_feature_map)
+        return X
+
+    def _fit(self,
+             X: pd.DataFrame,
+             y: pd.Series,
+             X_val: Optional[pd.DataFrame] = None,
+             y_val: Optional[pd.Series] = None,
+             time_limit: Optional[int] = None,
+             sample_weight=None,
+             verbosity=2,
+             **kwargs):
+        """The internal fit function
+
+        Parameters
+        ----------
+        X
```

Comment:
can doc be clearer about what X/X_val must look like?

Reply:
Removed this, it was copy-paste from TextPredictorModel